docs/db-migration/adr/007-dual-read-migration.md

```
@@ -0,0 +1,176 @@
+    override fun getComponentById(id: String): Component {
+        return if (sourceRegistry.isDbComponent(id))
+            dbResolver.getComponentById(id)
+        else
+            gitResolver.getComponentById(id)
+    }
+
+    override fun getAllComponents(): List<Component> {
+        val gitComponents = gitResolver.getAllComponents()
+            .filter { sourceRegistry.isGitComponent(it.id) }
+        val dbComponents = dbResolver.getAllComponents()
+            .filter { sourceRegistry.isDbComponent(it.id) }
```

markdownlint-cli2 (0.21.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)